packages/next/build/webpack/plugins/subresource-integrity-plugin.ts

```diff
@@ -0,0 +1,63 @@
+import { webpack, sources } from 'next/dist/compiled/webpack/webpack'
+import crypto from 'crypto'
+import { SUBRESOURCE_INTEGRITY_MANIFEST } from '../../../shared/lib/constants'
+
+const PLUGIN_NAME = 'SubresourceIntegrityPlugin'
+
+export type SubresourceIntegrityAlgorithm = 'sha256' | 'sha384' | 'sha512'
+
+export class SubresourceIntegrityPlugin {
+  constructor(private readonly algorithm: SubresourceIntegrityAlgorithm) {}
```

Comment:
Does this auto-assign? I couldn't find an example in the TypeScript docs for this 🤔

Reply:
It's a shorthand that I'd say you should only use if there's only a handful of positional arguments. Shouldn't use if you need to do any other work within the constructor.